File: examples/csharp/GiftGiver/GiftGiver/Models/Attendee.cs
using System;

namespace GiftGiver.Models
{
    public class Attendee : BaseEntity
    {
        public int AttendeeId { get; set; }
        public int VendorUserId { get; set; }
        public string RsvpAnswer { get; set; }
        public string Name { get; set; }
        public bool Awarded { get; set; }
    }
}

File: examples/ruby/sinatra/gift_giver/lib/attendee.rb
# frozen_string_literal: true

class Attendee < ActiveRecord::Base
  validates :name, :vendor_user_id, presence: true, uniqueness: true

  scope :awarded, -> { where(awarded: true) }
  scope :not_awarded, -> { where(awarded: false) }

  def self.random
    not_awarded.limit(1).order('random()').first
  end

  def award!
    update(awarded: true)
  end
end
